sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnBuilder.scala

```diff
@@ -181,6 +187,8 @@ private[columnar] object ColumnBuilder {
       case StringType => new StringColumnBuilder
       case BinaryType => new BinaryColumnBuilder
       case CalendarIntervalType => new IntervalColumnBuilder
+      case YearMonthIntervalType => new YearMonthIntervalColumnBuilder
+      case DayTimeIntervalType => new DayTimeIntervalColumnBuilder
```

Comment:
we can reuse `IntColumnBuilder` and `LongColumnBuilder`?

Reply:
+1